### PR TITLE
NEXT-7788 - Add config for disabling reviews

### LINKED ISSUE #852: NEXT-7788 - Disable ratings
EN:

It would be nice if you could disable the ratings completely.

DE:

Es wäre schön wenn man die Bewertungen komplett deaktivieren könnte.

### PR DESCRIPTION
**Pull request by sebi007**
### 1. Why is this change necessary?
Currently there is no way to disable reviews in the storefront except editing the templates.

### 2. What does this change do, exactly?
This change adds a config option. If this option is set, reviews are shown in the storefront. If not set, reviews are disabled.

### 4. Please link to the relevant issues (if any).
https://github.com/shopware/platform/issues/852

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [ ] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by mitelg** [comment]
@sebi007 you need to do a rebase :blush: there are conflicts

File: src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "sw-settings-listing": {
     "general": {
-      "mainMenuItemGeneral": "Produktliste",
```
Comment (mitelg) [inline src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json:4]: why did you changed this snippets? seems unrelated to disabling reviews

File: src/Core/Migration/Migration1589458026SetDefaultReviewConfig.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            'id' => Uuid::randomBytes(),
+            'configuration_key' => 'core.listing.showReview',
+            'configuration_value' => '{"_value": true}',
+            'created_at' => (new \DateTime())->format(Defaults::STORAGE_DATE_TIME_FORMAT),
+        ]);
+    }
+
+    public function updateDestructive(Connection $connection): void
+    {
+        // implement update destructive
```
Comment (mitelg) [inline src/Core/Migration/Migration1589458026SetDefaultReviewConfig.php]: could be deleted

File: src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig [hunk 3]
```diff
@@ -63,7 +63,7 @@
                     {% block component_product_box_info %}
                         <div class="product-info">
                             {% block component_product_box_rating %}
-                                {% if shopware.config.detail.showReviews %}
+                                {% if shopware.config.core.listing.showReview %}
```
Comment (mitelg) [inline src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig:66]: it seems there were plans for a product detail setting module? maybe you could talk about that with the design

File: src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig [hunk 4]
```diff
@@ -63,7 +63,7 @@
                     {% block component_product_box_info %}
                         <div class="product-info">
                             {% block component_product_box_rating %}
-                                {% if shopware.config.detail.showReviews %}
```
Comment (PaddyS) [inline src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig:66]: Not too sure about that, but I guess it would be nice to also remove the default configuration "showReviews", since it has no use now anymore.
`Shopware\Storefront\Framework\Twig\TemplateDataExtension#L147` that is.

@mitelg What do you think?

File: src/Storefront/Controller/ProductController.php [hunk 5]
```diff
@@ -157,11 +168,29 @@ public function saveReview(string $productId, RequestDataBag $data, SalesChannel
...
         return $this->renderStorefront('storefront/page/product-detail/review/review.html.twig', [
             'reviews' => $reviews,
             'ratingSuccess' => $request->get('success'),
         ]);
     }
+
+    /**
+     * @throws ReviewNotActiveExeption
+     */
+    protected function checkReviewsActive(SalesChannelContext $context): void
```
Comment (mitelg) [inline src/Storefront/Controller/ProductController.php]: please make this method private

**Comment by sebi007** [comment]
> @sebi007 you need to do a rebase 😊 there are conflicts

@mitelg Can you help me with rebasing?

File: src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json (hunk 1 above)
Comment (PaddyS) [reply src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json:4]: Design decision. It's more fitting this way

File: src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig (hunk 4 above)
Comment (mitelg) [reply src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig:66]: or the new config must have this key

File: src/Storefront/Controller/ProductController.php [hunk 6]
```diff
@@ -157,11 +168,29 @@ public function saveReview(string $productId, RequestDataBag $data, SalesChannel
...
+    {
+        $showReview = $this->systemConfigService->get('core.listing.showReview', $context->getSalesChannel()->getId());
+
+        if (
```
Comment (mitelg) [inline src/Storefront/Controller/ProductController.php]: check for false and throw the exception then.

Comment (stephan4p) [reply src/Storefront/Controller/ProductController.php]: We have write this function similar to `denyAccessUnlessLoggedIn()` and there is the same if and return struct

File: src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json (hunk 1 above)
Comment (mitelg) [reply src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json:4]: but the whole code only speaks of product listing. would be totally confusing, if there is now somehow a setting for the detail page

File: src/Core/Migration/Migration1589458026SetDefaultReviewConfig.php (hunk 2 above)
Comment (PaddyS) [reply src/Core/Migration/Migration1589458026SetDefaultReviewConfig.php]: You mean the comment?
The method itself has to remain. :-)

File: src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig (hunk 3 above)
Comment (PaddyS) [reply src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig:66]: It has to remain on the `listing` domain, so all they could do is change the key from `showReview` to `showReviews`.

Comment (PaddyS) [reply src/Storefront/Resources/views/storefront/component/product/card/box-standard.html.twig:66]: @mitelg Maybe add both conditions instead? Like `{% if shopware.config.detail.showReviews && shopware.config.core.listing.showReview%}` ?

File: src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json (hunk 1 above)
Comment (PaddyS) [reply src/Administration/Resources/app/administration/src/module/sw-settings-listing/snippet/de-DE.json:4]: It's also configured in a `listing.xml`, which is also confusing.
But we cannot change the name (e.g. from `listing.xml` to `products.xml` due to it breaking then) and the design wants it to be in this specific place.

Not really much to do here

File: src/Storefront/Controller/ProductController.php (hunk 5 above)
Comment (sebi007) [reply src/Storefront/Controller/ProductController.php]: Changed the method to private

**Comment by shyim** [comment]
Thanks for your Contribution! @sebi007 💙
